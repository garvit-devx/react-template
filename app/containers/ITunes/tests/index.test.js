/**
 *
 * Tests for ITunes container
 *
 *
 */

import React from 'react';
import { fireEvent } from '@testing-library/dom';
import { renderProvider } from '@utils/testUtils';
import { ITunesTest as ITunes } from '../index';

describe('<ITunes /> container tests', () => {
  let submitSpy;

  beforeEach(() => {
    submitSpy = jest.fn();
  });

  it('should render and match the snapshot', () => {
    const { baseElement } = renderProvider(<ITunes />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should call dispatchGetTracks when search form is submitted', () => {
    const SEARCH_TERM = 'ignite';

    const { getByTestId } = renderProvider(<ITunes dispatchGetTracks={submitSpy} />);
    const searchInput = getByTestId('search-input');
    const searchForm = getByTestId('search-form');

    fireEvent.change(searchInput, {
      target: { value: SEARCH_TERM }
    });

    fireEvent.submit(searchForm);
    expect(submitSpy).toBeCalledWith(SEARCH_TERM);
  });

  it('should display correct number of results', () => {
    const resultCount = 2;
    const results = [
      {
        trackName: 'ignite',
        artistName: 'Alan Walker'
      },
      {
        trackName: 'ignite',
        artistName: 'The Chainsmokers'
      }
    ];

    const { getByTestId } = renderProvider(<ITunes dispatchGetTracks={submitSpy} tracks={{ resultCount, results }} />);
    const totalResultsElement = getByTestId('total-results');

    expect(totalResultsElement).toHaveTextContent(resultCount);
  });

  it('should render exact number of TrackCard components as in the API response', () => {
    const resultCount = 2;
    const results = [
      {
        trackName: 'ignite',
        artistName: 'Alan Walker'
      },
      {
        trackName: 'ignite',
        artistName: 'The Chainsmokers'
      }
    ];

    const { getAllByTestId } = renderProvider(
      <ITunes dispatchGetTracks={submitSpy} tracks={{ resultCount, results }} />
    );
    expect(getAllByTestId('track-card').length).toBe(resultCount);
  });
});
