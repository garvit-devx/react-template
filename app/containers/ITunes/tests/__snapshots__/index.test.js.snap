// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ITunes /> container tests should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg e1qlpjed1 css-q20w81-MuiContainer-root"
    >
      <p
        class="css-lh9t18 egtqi0h0"
        data-testid="t"
      >
        iTunes Search
      </p>
      <form
        data-testid="search-form"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary css-djb9n3-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            data-testid="search-input"
            placeholder="Search any track"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-ihdtdm"
            >
              <span
                class="notranslate"
              >
                â€‹
              </span>
            </legend>
          </fieldset>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-secondary MuiButton-secondaryPrimary MuiButton-sizeMedium MuiButton-secondarySizeMedium MuiButton-root MuiButton-secondary MuiButton-secondaryPrimary MuiButton-sizeMedium MuiButton-secondarySizeMedium css-10yof0b-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Search
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </div>
</body>
`;
